### PR TITLE
EVA-3479 Remediate Lowercase nucleotide where both SVE has valid RS

### PR DESCRIPTION
**Pull request by nitin-ebi**

### CODE REVIEW

**Review by apriltuesday — APPROVED** [review]

**Review by tcezard — APPROVED** [review]
Mostly cosmetic comments.
Feel free to ignore them if needed

File: tasks/eva_3417/src/main/groovy/eva3417/BothSVEValidRSInvestigation.groovy [hunk 1]
```diff
@@ -0,0 +1,477 @@
...
+                continue
+            }
+        }
+
+        // deprecate SVE
+        logger.info("List of SVE to deprecate: " + gson.toJson(sveDeprecateList))
+        deprecateSVE(sveDeprecateList)
+
+        // insert SVE with new hash (after remediation)
+        insertSVEWithNewHash(sveInsertList, dbsnpSveClass)
```
[the remark this answers is not in the document]
Comment (nitin-ebi) [reply tasks/eva_3417/src/main/groovy/eva3417/BothSVEValidRSInvestigation.groovy:131]: added a comment and check for confirmation.

File: tasks/eva_3417/src/main/groovy/eva3417/BothSVEValidRSInvestigation.groovy [hunk 2]
```diff
@@ -0,0 +1,477 @@
...
+        // Even though both the RS are valid/exist, they have different start and only one of start matches with start in SVE
+        // Keep the SVE with RS whose start matches with start of the SVE (remediate if necessary)
+        // deprecate the SVE with RS whose start does not match with start of the SVE
+        List<SubmittedVariantEntity> sveInsertList = new ArrayList<>()
+        List<SubmittedVariantEntity> sveDeprecateList = new ArrayList<>()
+        for (BothRSValidCollision collision : bothRSValidCollisionList) {
+            SubmittedVariantEntity fileSVE = mapOfHashAndSVE.get(collision.getSveInFile())
+            SubmittedVariantEntity dbSVE = mapOfHashAndSVE.get(collision.getSveInDB())
+            ClusteredVariantEntity fileRS = mapOfAccessionAndCVE.get(fileSVE.getClusteredVariantAccession())
+            ClusteredVariantEntity dbRS = mapOfAccessionAndCVE.get(dbSVE.getClusteredVariantAccession())
```
[the remark this answers is not in the document]
Comment (nitin-ebi) [reply tasks/eva_3417/src/main/groovy/eva3417/BothSVEValidRSInvestigation.groovy]: updated

File: tasks/eva_3417/src/main/groovy/eva3417/BothSVEValidRSInvestigation.groovy [hunk 3]
```diff
@@ -0,0 +1,477 @@
...
+
+            // both SVE has same start, we can take it from any
+            Long sveStart = fileSVE.getStart()
+            if (sveStart == fileRS.getStart() && sveStart == dbRS.getStart()) {
+                logger.info("SVE_START_MATCHES_WITH_BOTH_RS")
+                continue
```
[the remark this answers is not in the document]
Comment (nitin-ebi) [reply tasks/eva_3417/src/main/groovy/eva3417/BothSVEValidRSInvestigation.groovy]: This is correct, the case is not present in investigation.csv. Added comment for clarification

File: tasks/eva_3417/src/main/groovy/eva3417/BothSVEValidRSInvestigation.groovy [hunk 4]
```diff
@@ -0,0 +1,477 @@
...
+            } else if (sveStart == fileRS.getStart() || sveStart == dbRS.getStart()) {
+                logger.info(sveStart == fileRS.getStart() ? "SVE_START_MATCHES_WITH_FILE_RS" : "SVE_START_MATCHES_WITH_DB_RS")
+                // assume fileSVE has the RS with correct start and needs to be kept
```
[the remark this answers is not in the document]
Comment (nitin-ebi) [reply tasks/eva_3417/src/main/groovy/eva3417/BothSVEValidRSInvestigation.groovy:110]: Added comment for clarification

**Pull request merged**